### ISSUE TITLE
Intg Test: structured events and expected errors not working

### ISSUE THREAD

**Issue by QuiteStochastic** [MinaProtocol/mina#8209]
the integration test framework is not quite detecting expected errors

**Comment by aneesharaines** [comment]
Has a fix but waiting on graphql ingress race condition to be fixed before testing.

**Comment by aneesharaines** [comment]
Looking like more of a refactor than a bug fix, @nholland94 to create a separate issue with details.  Only affects tests that expect errors like the send time payments tests.

**Comment by QuiteStochastic** [comment]
scope of this ticket has been expanded into https://github.com/MinaProtocol/mina/issues/8485
